packages/rum-core/src/domain/configuration.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Configuration,
+  DefaultPrivacyLevel,
+  display,
+  InitConfiguration,
+  isPercentage,
+  objectHasValue,
+  validateAndBuildConfiguration,
+} from '@datadog/browser-core'
+import { buildEnv } from '../boot/buildEnv'
+import { RumEventDomainContext } from '../domainContext.types'
+import { RumEvent } from '../rumEvent.types'
+
+export interface RumInitConfiguration extends InitConfiguration {
+  applicationId: string
+  beforeSend?: ((event: RumEvent, context: RumEventDomainContext) => void | boolean) | undefined
+  defaultPrivacyLevel?: DefaultPrivacyLevel | undefined
+}
+
+export type HybridInitConfiguration = Omit<RumInitConfiguration, 'applicationId' | 'clientToken'>
+
+export const DEFAULT_RUM_CONFIGURATION = {
+  maxActionsPerMinute: 3000,
+
+  replaySampleRate: 100,
+  allowedTracingOrigins: [] as Array<RegExp | string>,
+  trackInteractions: false,
+  trackViewsManually: false,
+  defaultPrivacyLevel: DefaultPrivacyLevel.MASK_USER_INPUT as DefaultPrivacyLevel,
+}
+
+export type RumConfiguration = Configuration &
+  typeof DEFAULT_RUM_CONFIGURATION & {
+    applicationId: string
+
+    actionNameAttribute: string | undefined
+  }
+
+export function validateAndBuildRumConfiguration(
```

Comment:
I tried a different implementation [approach](https://github.com/DataDog/browser-sdk/blob/aymeric/revamp-configuration-rum-refacto/packages/rum-core/src/domain/configuration.ts):
My pros:
- Shorter
- Isolate init conf check at the top of the function
- No configuration mutation
- Less objective but avoid the RumConfiguration type union which I found a bit complicated 
Cons 
- Init config undefined check is done twice
- The default config might be a bit more hard to read

Let me know what you think. If we choose to do it. It can be done as a follow up to not block the v4.